jwst/ramp_fitting/tests/test_ramp_fit.py

```diff
@@ -836,11 +836,66 @@ def test_miri_no_good_pixel():
     model1.groupdq[:, :, :, :] = DO_NOT_USE
 
     image_info, int_model, opt_model, gls_opt_model = ramp_fit(
-        model1, 1024 * 30000., True, rnoise, gain, 'OLS', 'optimal', 'none', test_dq_flags)
+        model1, 1024 * 30000., True, rnoise, gain, 'OLS', 'optimal', 'none', dqflags.pixel)
 
     assert image_info is None
 
 
+def test_zero_frame_usage():
+    """
+    Test using ZEROFRAME data for fully saturated ramps, with three different
+    ramps.
+
+    Ramp descriptions:
+    0. One good group, then saturated for groups 1 and on.
+        - Neither the data, nor the groupdq will be touched
+    1. Zero good groups, since fully saturated, 0. data in ZEROFRAME.
+        - Neither the data, nor the groupdq will be touched
+    2. Zero good groups, since fully saturated, with good data in ZEROFRAME.
+        - The zero group will be replaced and the 
+    """
+    nints, ngroups, nrows, ncols = 1, 5, 1, 3
+    deltatime = 3.
+    model, gdq, rnoise, pixdq, err, gain = setup_small_cube(
+        ngroups, nints, nrows, ncols, deltatime)
+
+    # XXX ZEROFRAME correction; will ultimately be fixed in JP-2506.
+    correct_shape = (nints, nrows, ncols)
+    if model.zeroframe.shape != correct_shape:
+        model.zeroframe = np.zeros(correct_shape, dtype=model.data.dtype)
+
+    base_slope = 50.0
+    base_arr = [10000. + k * base_slope for k in range(ngroups)]
+    base_ramp = np.array(base_arr, dtype=np.float32)
+
+    # Make all ramps the same.  The only difference will be 
+    model.data[0, :, 0, 0] = base_ramp
+    model.data[0, :, 0, 1] = base_ramp
+    model.data[0, :, 0, 2] = base_ramp
+    data_check = model.data.copy()
+    data_check[0, 0, 0, 2] /= 2.
+
+    model.zeroframe[0, 0, 0] = model.data[0, 0, 0, 0] / 2.
+    model.zeroframe[0, 0, 1] = 0.
+    model.zeroframe[0, 0, 2] = model.data[0, 0, 0, 2] / 2.
+
+    # Set all groups to DO_NOT_USE
+    model.groupdq[:, :, :, :] = SATURATED
+    model.groupdq[0, 0, 0, 0] = GOOD
+    groupdq_check = model.groupdq.copy()
+    groupdq_check[0, 0, 0, 2] = GOOD
+
+    use_zeroframe_for_saturated_ramps(model, dqflags.pixel)
+
+    # Make sure the group 0 info changed for ramp 2 in the data
+    # and groupdq arrays.
+    tol = 1.e-5
+    np.testing.assert_allclose(model.data, data_check, tol)
+
+    check = np.array(groupdq_check, dtype=np.uint8)
+    np.testing.assert_array_equal(model.groupdq, groupdq_check)
```

Comment:
Is "check" used ?

Reply:
No.  It was a relic from a previous iteration of this test.  I have removed it.